packages/expo-cli/src/credentials/views/IosPushCredentials.ts

```diff
@@ -340,6 +347,13 @@ async function selectPushCredFromList(
   accountName: string,
   options: ListOptions = {}
 ): Promise<IosPushCredentials | IosAppCredentials | null> {
+  if (ctx.nonInteractive) {
+    throw new CommandError(
+      'NON_INTERACTIVE',
+      "Start the CLI without the '--non-interactive' flag to select a push notification credential."
+    );
+  }
+
```

Comment:
```suggestion
  if (ctx.nonInteractive) {
    throw new CommandError(
      'NON_INTERACTIVE',
      "Start the CLI without the '--non-interactive' flag to select a push notification credential."
    );
  }

```

if it's happening then handle it outside that function